### PR TITLE
[DAGCombiner] Simplifying `{si|ui}tofp` when only signbit is needed    

### PR DESCRIPTION
**Pull request by goldsteinn**
- **[DAGCombiner] Add tests for simplifying `{si|ui}tofp`; NFC**
- **[DAGCombiner] Simplifying `{si|ui}tofp` when only signbit is needed**

If we only need the signbit `uitofp` simplified to 0, and `sitofp`
simplifies to `bitcast`.

### CODE REVIEW

File: llvm/test/CodeGen/X86/int-to-fp-demanded.ll [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+  %i = bitcast float %f to i32
+  %r = and i32 %i, 2147483648
+  ret i32 %r
+}
+
+define i32 @sitofp_signbit_only_okay_width(i16 %i_in) {
+; X86-LABEL: sitofp_signbit_only_okay_width:
+; X86:       # %bb.0:
+; X86-NEXT:    subl $8, %esp
+; X86-NEXT:    .cfi_def_cfa_offset 12
```
Comment (phoebewang) [inline llvm/test/CodeGen/X86/int-to-fp-demanded.ll:30]: Add `nounwind` to suppress the CFI directives.

**Review by phoebewang — APPROVED** [review]
LGTM.

**Comment by jayfoad** [comment]
Do you really have to duplicate the same logic in `SimplifyMultipleUseDemandedBits` and `SimplifyDemandedBits`?

**Comment by goldsteinn** [comment]
> Do you really have to duplicate the same logic in `SimplifyMultipleUseDemandedBits` and `SimplifyDemandedBits`?

I can make it a helper. My feeling was its only 1-2 lines of code so not really needed.

**Review by RKSimon — APPROVED** [review]
LGTM - cheers

**Comment by RKSimon** [comment]
@goldsteinn The b60cf84e0965ac12b83494f803ea0dd6dec0db77 commit doesn't appear to match this PR?

**Comment by RKSimon** [comment]
Actual commit was 353fbeb0a294d2c7cef6d88607fa0fd50ee81462

**Comment by goldsteinn** [comment]
Bah, sorry, committed both at once, got messed up :/ rebasing the other two PRs shortly.

**Pull request closed without merging**